src/MongoDB.Driver.Core/Core/Servers/Server.cs

```diff
@@ -240,7 +240,8 @@ private void OnDescriptionChanged(object sender, ServerDescriptionChangedEventAr
                 _connectionPool.Clear();
             }
 
-            if (_descriptionChangedEventHandler != null)
+            var shouldServerDescriptionChangedEventBePublished = !e.OldServerDescription.SdamEquals(e.NewServerDescription);
+            if (shouldServerDescriptionChangedEventBePublished && _descriptionChangedEventHandler != null)
```

Comment:
I think the `Event` should *always* be raised.

It is up to the receiver to decide what to do (or not do).

Reply:
I see now what confused me...

We raise this event TWICE in this method, once using our own event mechanism and once using C# events.

Your change is ONLY suppressing the FIRST of the two events, so we are in fact ALWAYS raising the SECOND event, which matches what I expected in my first comment in this conversation.